test/e2e/allocator/pod_termination_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2023 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package allocator
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	pb "agones.dev/agones/pkg/allocation/go"
+	agonesv1 "agones.dev/agones/pkg/apis/agones/v1"
+	helper "agones.dev/agones/test/e2e/allochelper"
+	e2e "agones.dev/agones/test/e2e/framework"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	retryInterval = 1 * time.Second
+	retryTimeout  = 30 * time.Second
+)
+
+func TestAllocatorAfterDeleteReplica(t *testing.T) {
+	ctx := context.Background()
+
+	// get allocator pods
+	list, err := helper.GetAgonesAllocatorPods(ctx, framework)
+	assert.NoError(t, err)
+
+	grpcClient, err := helper.GetAllocatorClient(ctx, t, framework)
+	require.NoError(t, err, "Could not initialize rpc client")
+
+	// create fleet
+	flt, err := helper.CreateFleet(ctx, framework.Namespace, framework)
+	if !assert.Nil(t, err) {
+		return
+	}
+	framework.AssertFleetCondition(t, flt, e2e.FleetReadyCount(flt.Spec.Replicas))
+
+	var response *pb.AllocationResponse
+	request := &pb.AllocationRequest{
+		Namespace:                    framework.Namespace,
+		PreferredGameServerSelectors: []*pb.GameServerSelector{{MatchLabels: map[string]string{agonesv1.FleetNameLabel: flt.ObjectMeta.Name}}},
+		Scheduling:                   pb.AllocationRequest_Packed,
+		Metadata:                     &pb.MetaPatch{Labels: map[string]string{"gslabel": "allocatedbytest"}},
+	}
+
+	// delete 2 of the allocators
+	for _, pod := range list.Items[1:] {
+		err = helper.DeleteAgonesAllocatorPod(ctx, pod.ObjectMeta.Name, framework)
+		require.NoError(t, err, "Could not delete allocator pod")
+	}
+
+	response, err = grpcClient.Allocate(context.Background(), request)
+	logrus.Info(response)
+	helper.ValidateAllocatorResponse(t, response)
+	require.NoError(t, err, "Failed grpc allocation request")
+
+	// Wait and make another allocation call once we know the draining time has passed
+	_ = wait.PollImmediate(retryInterval, retryTimeout, func() (bool, error) {
```

Comment:
The point of the wait.PollImmediate was to do the allocation inside this function (you can just keep returning false), multiple times. What you have here is a sleep.